presto-iceberg/src/main/java/com/facebook/presto/iceberg/IcebergNativeMetadata.java

```diff
@@ -376,4 +379,13 @@ public void unregisterTable(ConnectorSession clientSession, SchemaTableName sche
     {
         catalogFactory.getCatalog(clientSession).dropTable(toIcebergTableIdentifier(schemaTableName, catalogFactory.isNestedNamespaceEnabled()), false);
     }
+
+    protected Optional<String> getDataLocationBasedOnWarehouseDataDir(SchemaTableName schemaTableName)
+    {
+        if (!catalogType.equals(HADOOP)) {
+            return Optional.empty();
+        }
+        return Optional.ofNullable(warehouseDataDir.map(base -> base + schemaTableName.getSchemaName() + "/" + schemaTableName.getTableName())
+                .orElse(null));
```

Comment:
`Optional.ofNullable` is redundant here

```suggestion
        return warehouseDataDir.map(base -> base + schemaTableName.getSchemaName() + "/" + schemaTableName.getTableName());
```

Reply:
Nice catch, fixed!